src/node_zlib.cc

```diff
@@ -254,11 +253,22 @@ class ZCtx : public AsyncWrap {
             ctx->err_ = Z_NEED_DICT;
           }
         }
+        if (ctx->strm_.avail_in >= 2 && ctx->mode_ == GUNZIP) {
```

Comment:
Magic constants need an explaining comment and should preferably be true, well-named constants.